src/WebJobs.Script.Grpc/Extensions/RpcSharedMemoryDataExtensions.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
+using Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc.Extensions
```

Comment:
Adding to this file but this is more of a general comment. This is not specific to Grpc, so we should probably not have this functionality under this namespace. I'm also not really sure about the Rpc prefix on the new types.

Reply:
The overall shared memory data transfer is not specific to Grpc, you are right.
However, the `RpcSharedMemory` type *is* specific to Grpc (it's defined in `FunctionRpc.proto`) as its a means to provide metadata about the shared memory maps between worker/host that are communicating over Grpc (for control messages, not the actual data.) That's why we've kept this under the Grpc namespace. Other more general components (like `SharedMemoryManager`) are *not* in this namespace.

Reply:
That makes sense. Provided the feedback as I thought I had seen some additional types under this namespace, but if not, please disregard.